### ISSUE TITLE
gluing webassembly and javascript together

### ISSUE THREAD

**Issue by TooManyBees** [brooklynjs/brooklynjs.github.io#503]
## BrooklynJS Talk Submission

**Talk Title:** gluing webassembly and javascript together

**Talk Description:**

There are a lot of WebAssembly guides focused on writing libraries in [some language Not Javascript] and ending up with a bundle of JavaScript that you can call from your browser or Node and it *just works* and you might never know WebAssembly was involved unless you peeked at the source files. While this is actual literal magic, it leaves a lot unexplained, and this mystery left me hesitant to try WebAssembly until I figured out what the magic compilers were doing and why.

This talk is about the boundary region between a blob of WebAssembly bytecode and the Javascript environment that calls it. Specifically:
* a quick summary of WABT, the [web assembly binary toolkit](https://github.com/WebAssembly/wabt)
* how WebAssembly exports functions for Javascript to call
* how WebAssembly imports and calls foreign functions
* what data WebAssembly can and can't represent
* what data can be shared across WebAssembly and Javascript environments
* how to represent complex structs in allocated memory
* how to deal with strings
* an example that wraps all of that ^^ together

In other words, I'm going to talk about all the glue code that a programmer needs to implement for WebAssembly to be conveniently usable, which should help explain what the magic bundlers are doing behind the scenes.

**Name**: Jesse Kipp

**Twitter Handle (optional)**: @toomanybees

**Website (optional):** toomanybees.com

**Months Available to Speak:** September, November, December

**Comment by paladique** [comment]
@TooManyBees your talk has been accepted for this month's lineup! Please email organizers@brooklynjs.com for details and your speaker ticket!
